Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -925,8 +926,9 @@ LayoutUnit RenderFlexibleBox::computeMainSizeFromAspectRatioUsing(const RenderBo
     if (crossSizeLength.isFixed())
         crossSize = LayoutUnit(crossSizeLength.value());
     else if (crossSizeLength.isAuto()) {
-        ASSERT(childCrossSizeShouldUseContainerCrossSize(child));
-        crossSize = computeCrossSizeForChildUsingContainerCrossSize(child);
+        std::optional<LayoutUnit> containerCrossSize = childCrossSizeShouldUseContainerCrossSize(child); 
+        ASSERT(containerCrossSize);
+        crossSize = computeCrossSizeForChildUsingContainerCrossSize(child, containerCrossSize);
```

Comment:
This is still a bit problematic as we could very well end up with a std:null_opt here and it's too late to fallback to a non-zero height.